src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php

```diff
@@ -68,9 +116,16 @@ public function applyToItem(QueryBuilder $queryBuilder, QueryNameGeneratorInterf
      * @param string       $originAlias             the current entity alias (first o, then a1, a2 etc.)
      * @param string       $relationAlias           the previous relation alias to keep it unique
      * @param bool         $wasLeftJoin             if the relation containing the new one had a left join, we have to force the new one to left join too
+     * @param int          $joinsNumber             the number of joins
+     *
+     * @throws RuntimeException when the max number of joins has been reached
      */
-    private function joinRelations(QueryBuilder $queryBuilder, string $resourceClass, array $propertyMetadataOptions = [], string $originAlias = 'o', string &$relationAlias = 'a', bool $wasLeftJoin = false)
+    private function joinRelations(QueryBuilder $queryBuilder, string $resourceClass, array $propertyMetadataOptions = [], string $originAlias = 'o', string &$relationAlias = 'a', bool $wasLeftJoin = false, $joinsNumber = 0)
     {
+        if ($joinsNumber > $this->maxJoins) {
+            throw new RuntimeException('Too many joined relations, might be an infinite loop! Check out serialization groups!');
```

Comment:
My suggestion:

Total number of joined relations has exceeded the specified maximum. Raise the limit if necessary.